### PR TITLE
Feature/workers and research

### PR DESCRIPTION
**Pull request by gdscheele**

### CODE REVIEW

File: src/main/game/WorkerResearchObservable.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package game;
+
+import game.resources.Resource;
```
Comment (karthshen) [inline src/main/game/WorkerResearchObservable.java]: lower case i for interface

File: src/main/game/entities/Player/Player.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import game.ResearchObservable;
+import game.entities.workers.workerManagement.exceptions.WorkerLimitExceededException;
+import game.entities.workers.workerTypes.WorkerTypeEnum;
+import game.gameboard.Location;
+
+public class Player{
+
+    private WorkerManager workerManager;
+    private ResearchObservable researchObservable;
+    private int playerId;
```
Comment (karthshen) [inline src/main/game/entities/Player/Player.java]: comment it out

Comment (gdscheele) [inline src/main/game/entities/Player/Player.java]: removed this file

File: src/main/game/WorkerResearchObservable.java (hunk 1 above)
Comment (gdscheele) [inline src/main/game/WorkerResearchObservable.java]: kk

File: src/main/game/entities/workers/workerManagement/WorkerFactory.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            case POWER_GENERATOR:
+                return new PowerGenerator(entityId, workerProductionStatistics.get(workerType), location);
+            case WORKER_GENERATOR:
+                return new WorkerGenerator(entityId, workerProductionStatistics.get(workerType), location);
+            case RESEARCH_GENERATOR:
+                return new ResearchGenerator(entityId, workerProductionStatistics.get(workerType), location);
+            case SOLDIER_GENERATOR:
+                return new SoldierGenerator(entityId, workerProductionStatistics.get(workerType), location);
+        }
+        return null;
```
Comment (Aclark2089) [inline src/main/game/entities/workers/workerManagement/WorkerFactory.java]: Throw an exception for a worker type not being found

**Pull request merged**